### PR TITLE
Fixed calling the datagrid query three times instead of one if there is no result available.

### PR DESCRIPTION
**Pull request by pulzarraider**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Would not be better to avoid query repetition inside the datagrid class?

**Comment by Soullivaneuh** [comment]
Note: I didn't bump 2.4 version for now. If this bug exists on 2.3, this should PR should be done here instead IMO.

File: Resources/views/CRUD/list_outer_rows_list.html.twig [hunk 1]
```diff
@@ -9,7 +9,7 @@ file that was distributed with this source code.
 
 #}
 
-{% for object in admin.datagrid.results %}
+{% for object in datagrid_results %}
```
Comment (core23) [inline Resources/views/CRUD/list_outer_rows_list.html.twig]: Are you sure that the ``datagrid_result`` exists here? You could override ``base_list.html.twig`` without definging ``datagrid_result``.

File: Resources/views/CRUD/list_outer_rows_mosaic.html.twig [hunk 2]
```diff
@@ -19,7 +19,7 @@ This template can be customized to match your needs. You should only extends the
 <tr>
     <td colspan="{{ admin.list.elements|length - (app.request.isXmlHttpRequest ? (admin.list.has('_action') + admin.list.has('batch')) : 0) }}">
         <div class="row">
-            {% for object in admin.datagrid.results %}
+            {% for object in datagrid_results %}
```
Comment (core23) [inline Resources/views/CRUD/list_outer_rows_mosaic.html.twig]: Same here

**Comment by core23** [comment]
Ping @pulzarraider

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**